pims/base_frames.py

```diff
@@ -528,3 +528,268 @@ def process(img_or_iterable, *args, **kwargs):
                        "other objects, its behavior is "
                        "unchanged.\n\n") + process.__doc__
     return process
+
+
+class FramesDimension(object):
+    """
+    This class defines a dimension for use in the Multidimensional base class.
+
+    Parameters
+    ----------
+    name : string, dimension name
+    size : int, size of dimension
+    aggregate : boolean, default False
```

Comment:
Things that are optional should be `name : type, optional`  with the default given in the docstring.